### PR TITLE
add user parameters for setting robot state

### PR DESCRIPTION
**Pull request by OmkarKabadagi5823**
Fixes adamheins/pyb_utils#1

## Features added:
1. User debug parameters for all the joints of robot arm (7 joints, min: -2pi, max: 2pi, default: 0).
2. User debug parameter to set collision margin (min: 0; max: 0.2; default: 0.01).
3. Update robot in GUI only if the robot is not within collision margin with any object.
4. Display a short message on the GUI if robot is within collision margin. Disappears in 0.2 secs once robot is out of collision.

### User Debug Parameters
Allows to set arbitrary joint values for the 7 robot joints via GUI sliders. Also allows to set collision margin from the GUI.
image

### Message when robot in collision margin
Invalid joint configurations can be set using the sliders but only valid configurations are updated on the GUI. A Message on GUI indicates robot is within collision margin with an object. The message disappears in 0.2 secs once the robot is out of collision.
image

### CODE REVIEW

File: scripts/collision_detection_example.py [hunk 1]
```diff
@@ -46,6 +46,21 @@ def load_environment(client_id):
     }
     return bodies
 
+def create_user_debug_params(gui_id):
+    return {
+        'collision_margin': pyb.addUserDebugParameter('collision_margin', 0, 0.2, 0.01, physicsClientId=gui_id),
+        'lbr_iiwa_joint1': pyb.addUserDebugParameter('lbr_iiwa_joint1', -2*np.pi, 2*np.pi, 0, physicsClientId=gui_id),
```
Comment (adamheins) [inline scripts/collision_detection_example.py:52]: Could you populate the joint entries here with a loop to make it a bit cleaner?

File: scripts/collision_detection_example.py [hunk 2]
```diff
@@ -73,21 +91,38 @@ def main():
         [(link7, ground), (link7, cube1), (link7, cube2), (link7, cube3)],
     )
 
+    robot_id = bodies["robot"]
+
     while True:
         # compute shortest distances for a random configuration
-        q = np.pi * (np.random.random(7) - 0.5)
+        q = np.array([
```
Comment (adamheins) [inline scripts/collision_detection_example.py:98]: Can we make this a function, say `read_robot_state`, which populates `q` with a loop over the debug params?

**Review by adamheins — CHANGES_REQUESTED** [review]
I like the interactivity this PR adds. Aside from the couple specific comments I made, for consistency can you format the file using [Black](https://github.com/psf/black)?

File: scripts/collision_detection_example.py (hunk 1 above)
Comment (OmkarKabadagi5823) [reply scripts/collision_detection_example.py]: Fixed

File: scripts/collision_detection_example.py (hunk 2 above)
Comment (OmkarKabadagi5823) [reply scripts/collision_detection_example.py]: I have moved reading debug params to `read_robot_state` but I am still manually setting the state array. The reason is that reading them over a loop was making the robot state change in GUI jittery

**Review by adamheins — APPROVED** [review]

**Comment by adamheins** [comment]
LGTM.

**Pull request merged**